src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -9651,8 +9653,11 @@ private void WriteSmiParameter(SqlParameter param, int paramIndex, bool sendDefa
                 typeCode = MetaDataUtilsSmi.DetermineExtendedTypeCodeForUseWithSqlDbType(metaData.SqlDbType, metaData.IsMultiValued, value, null);
             }
 
-            var sendDefaultValue = sendDefault ? 1 : 0;
-            SqlClientEventSource.Log.AdvancedTraceEvent("<sc.TdsParser.WriteSmiParameter|ADV> {0}, Sending parameter '{1}', default flag={2}, metadata:{3}", ObjectID, param.ParameterName, sendDefaultValue, metaData.TraceString(3));
+            if (advancedTraceIsOn)
```

Comment:
Why not `if (SqlClientEventSource.Log.IsAdvancedTraceOn())` directly?

Reply:
This call is inside a loop, once for each parameter. I didn't see the point repeatedly checking since it's highly unlikely to have the trace flag turned on in the middle. It can be cheeked directly if you prefer.

Reply:
If there's no perf impact, I'd rather prefer keeping it straightforward.

Reply:
The perf impact is whatever it takes to call `SqlClientEventSource.Log.IsAdvancedTraceOn()` multiplied by the number of parameters. I'm not sure how the trace flags are actually checked or what overhead they have. There's some perf impact to going the same thing N times instead of once and storing the result but I can't say I know what it is or whether it's important. Based on that, do you want me to make the change?

Reply:
I lean a bit towards keeping it as-is for the perf benefit in this case... even though it's probably pretty minor.

Reply:
I approved this PR. I agree with David. The effect on the perf is pretty minor.